### PR TITLE
Switch to variable concepts, and other things

### PR DESCRIPTION
**Pull request by ericniebler**
~~Don't merge yet. There is a high likelihood that 830e9fd stomped some changes. I need to investigate.~~

OK, I fixed a bad table and reinstated 00d9f8d which got lost in the shuffle. I also fixed more occurrences of "an `Range`" as editorial. This is ready for its close-up.

### CODE REVIEW

**Comment by ericniebler** [comment]
(One more forced update to fix a commit message.)

File: concepts.tex [hunk 1]
```diff
@@ -76,13 +76,12 @@
 \enterexample
 \begin{codeblock}
 template <class T>
-concept bool C() {
-  return requires(T a, T b, const T c, const T d) {
+concept bool C =
```
Comment (CaseyCarter) [inline concepts.tex]: Nice catch!

Comment (CaseyCarter) [inline concepts.tex]: Then again, I suppose you grepped for `()` and found *everything*

File: concepts.tex [hunk 2]
```diff
@@ -132,198 +131,126 @@
...
-    return @\seebelow@;
-  }
+  concept bool Boolean = @\seebelow@;
 
   // \ref{concepts.lib.compare.equalitycomparable}, EqualityComparable:
   template <class T, class U>
-  concept bool WeaklyEqualityComparable() {
-    return @\seebelow@;
-  }
+  concept bool With = @\seebelow@;
```
Comment (CaseyCarter) [inline concepts.tex]: Errr, `With`?

File: concepts.tex [hunk 3]
```diff
@@ -443,26 +363,25 @@
 \indexlibrary{\idxcode{Common}}%
 \begin{itemdecl}
 template <class T, class U>
-concept bool Common() {
-  return CommonReference<const T&, const U&>() &&
+concept bool Common =
+  CommonReference<const T&, const U&> &&
     requires(T (&t)(), U (&u)()) {
```
Comment (CaseyCarter) [inline concepts.tex]: We really should -1 (outdent) the rest of this definition.

File: concepts.tex [hunk 4]
```diff
@@ -928,23 +830,21 @@
   };
 
 template <class T, class... Args>
-concept bool Constructible() {
-  return @\xname{ConstructibleObject}@<T, Args...> ||
-    @\xname{BindableReference}@<T, Args...>;
-}
+concept bool Constructible =
+  @\xname{ConstructibleObject}@<T, Args...> ||
```
Comment (CaseyCarter) [inline concepts.tex]: *hsssssst* disjunction, we hates it.

File: iterators.tex [hunk 5]
```diff
@@ -70,138 +70,84 @@
...
-  }
   template <class F, class I>
-  concept bool IndirectRegularInvocable() {
-    return @\seebelow@;
-  }
-  template <class F, class I1, class I2>
-  concept bool IndirectRegularInvocable() {
-    return @\seebelow@;
-  }
+  concept bool IndirectRegularUnaryInvocable = @\seebelow@;
```
Comment (CaseyCarter) [inline iterators.tex]: That'll teach people to accuse us of writing long concept names!

File: concepts.tex (hunk 4 above)
Comment (CaseyCarter) [inline concepts.tex]: (No change requested, just recoiling in horror from the disjunction.)

**Comment by ericniebler** [comment]
Better now?

File: concepts.tex [hunk 6]
```diff
@@ -443,26 +363,25 @@
...
+concept bool Common =
+  CommonReference<const T&, const U&> &&
+  requires(T (&t)(), U (&u)()) {
+    typename common_type_t<T, U>;
+    typename common_type_t<U, T>;
+    requires Same<common_type_t<U, T>, common_type_t<T, U>>;
+    common_type_t<T, U>(t());
+    common_type_t<T, U>(u());
+    requires CommonReference<add_lvalue_reference_t<common_type_t<T, U>>,
+                              common_reference_t<add_lvalue_reference_t<const T>,
```
Comment (CaseyCarter) [inline concepts.tex]: This line has one space too much indent.

Comment (CaseyCarter) [inline concepts.tex]: Other than that, it's great. Merge at will.

**Comment by ericniebler** [comment]
Merged!

**Pull request merged**